src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/InputEmptyCase.java

```diff
@@ -0,0 +1,36 @@
+////////////////////////////////////////////////////////////////////////////////
+// Input test file for testing empty LITERAL_CASE.
+// Created: 2017
+////////////////////////////////////////////////////////////////////////////////
+package com.puppycrawl.tools.checkstyle.checks.blocks;
+
+class InputEmptyCase
+{
+    void method1(int a) {
+        switch (a) {}
+        switch (a) {case 1: ; } // no violation
+        switch (a) {case 1:{}}  // 1 violation
+        switch (a) {
+            case 1:
+        }
+        switch (a) {
+            case 1: // 1 violation
+            {}
+        }
+        switch (a) {
+            case 1: // 1 violation if checking statements
+            {// none if checking text
+            }
+        }
+    }
+
+    public void method2(char c) {
+        switch(c) { case 0: } // no violation
+        switch(c) { case 0: {} } // 1 violation
+        switch(c) { case 0: method1(0); {} } // no violation
+        switch(c) { case 0: case 1: {} } // 1 violation
+        switch(c) { case 0: {} case 1: {// 2 violations if checking statements
+        }
+        }
+    }
+}
```

Comment:
https://github.com/checkstyle/checkstyle/issues/3839#issuecomment-280785109

> This Check care about block that goes RIGHT AFTER defined token.

yes. This is why `a3` has no violation in issue example. Token is `case` and `test()` appears right after it, not the block.

> This check should validate only first { } in your case and raise violation.

yes. This is the difference between `a2` and `a3` in issue example.

> This Check does not validate sequential blocks {}, you make put blocks at any possible place in code.

yes. This is similar to `a3` in issue example.
If user has code as `{} {}`, we only expect 1 violation, not 2.